### PR TITLE
Game Engine Assignment 1

### PR DESCRIPTION
**Pull request by HAOlleik**
# Description
PR contains the engine component.

# Steps for Running
1- Run the command `g++ src/Main.cpp src/GameEngine.cpp -o gameEngine.out` for compilation
2- Run the command `./gameEngine.out` to run the program
3- Enter input, input is the action and they are based on the way they are mentioned in pdf (loadmap/validatemap/...)

### CODE REVIEW

**Review by Yason287 — COMMENTED** [review]
Looks good

File: src/GameEngine.h [hunk 1]
```diff
@@ -1,28 +1,53 @@
...
+
+extern std::map<State, std::map<Action, State>> mapStateToActions;
+extern std::map<std::string, Action> actionToString;
+
+void inputToLower(char* input);
+
 class GameEngine
 {
 private:
     State state;
```
Comment (daxsis) [inline src/GameEngine.h:42]: should be a pointer

**Review by Zain-MKhan — APPROVED** [review]

**Review by uzairali245 — APPROVED** [review]
Lgtm

**Review by Yason287 — COMMENTED** [review]
Looks good

**Review by daxsis — APPROVED** [review]

**Comment by daxsis** [comment]
how to run is actually even easier than what you wrote

- `make clean`
- `make all`
- `bin/conquest.bin`
That's it

**Pull request merged**